src/Neovim.jl

```diff
@@ -9,38 +9,41 @@ export Buffer, Tabpage, Window
 export reply_result, reply_error
 
 # types that have api methods (nvim itself + api defined types)
-abstract NvimObject
+abstract type NvimObject end
 
 include("client.jl")
 include("api_gen.jl")
 include("interface.jl")
 include("plugin_host.jl")
 
 # too inconvenient api to supply handler here?
-function nvim_connect(path::ByteString, args...)
+function nvim_connect(path::String, args...)
     s = connect(path)
     NvimClient(s, s, args...)
 end
 
 nvim_env(args...) = nvim_connect(ENV["NVIM_LISTEN_ADDRESS"], args...)
 
 function nvim_spawn(args...; cmd=`nvim --embed`)
-    output, input, proc = readandwrite(cmd)
-    (NvimClient(input, output, args...), proc)
+    # TODO(smolck): Just use the Process handle instead of p.in and p.out
+    #output, input, proc = readandwrite(cmd)
+    # (NvimClient(p, args...), proc)
+    p = open(cmd, "r+")
+    (NvimClient(p.in, p.out, args...), p)
 end
 
 function nvim_child(args...)
     # make stdio private. Reversed since from nvim's perspective
-    input, output = STDOUT, STDIN
+    input, output = stdout, stdin
     if haskey(ENV, "NEOVIM_JL_DEBUG")
         debug = open(ENV["NEOVIM_JL_DEBUG"], "w")
         redirect_stdout(debug)
         redirect_stderr(debug)
     else
-        redirect_stdout()
-        redirect_stderr()
+        # redirect_stdout()
```

Comment:
is this WIP?

Reply:
Good catch, I think those can be safely uncommented now. I thought they were causing some issues but I think the cause was actually something else.